test/api_save_product_test.dart

```diff
@@ -329,23 +329,112 @@ Like that:
+            ..setValue(Nutrient.energyKJ, perSize, ENERGY + i)
+            ..setValue(Nutrient.carbohydrates, perSize, CARBOHYDRATES + i)
+            ..setValue(Nutrient.proteins, perSize, PROTEINS + i)
+            ..setValue(Nutrient.vitaminB12, perSize, VITAMIN_B12 + i)
+            ..setValue(Nutrient.fat, perSize, FAT + i);
+
+          final Product newProduct = Product(
+            barcode: BARCODE,
+            productName: PRODUCT_NAME,
+            nutrimentDataPer: nutrimentDataPer,
+            servingSize: "30g",
+            nutriments: nutriments,
+          );
+
+          final Status status = await OpenFoodAPIClient.saveProduct(
+            USER,
+            newProduct,
+            uriHelper: uriHelper,
+          );
+
+          expect(status.status, 1);
+          expect(status.statusVerbose, 'fields saved');
+
+          final ProductResultV3 result = await OpenFoodAPIClient.getProductV3(
+            ProductQueryConfiguration(
+              BARCODE,
+              language: OpenFoodFactsLanguage.ENGLISH,
+              version: ProductQueryVersion.v3,
+            ),
+            user: USER,
+            uriHelper: uriHelper,
+          );
+
+          expect(result.status, ProductResultV3.statusSuccess);
+          expect(result.barcode, BARCODE);
+
+          final Product? searchedProduct = result.product;
+          if (searchedProduct == null) {
+            fail('Saved product not found');
+          }
 
-      const PerSize perSize = PerSize.oneHundredGrams;
-      for (int i = 2; i >= 0; i--) {
+          expect(searchedProduct.barcode, BARCODE);
+          expect(searchedProduct.productName, PRODUCT_NAME);
+          expect(searchedProduct.nutrimentDataPer, nutrimentDataPer);
+
+          final searchedNutriments = searchedProduct.nutriments;
+          expect(searchedNutriments, isNotNull);
+          if (searchedNutriments == null) {
+            fail('Nutrients are null');
+          }
+          for (final Nutrient nutrient in nutrients) {
+            final double? actual =
+                searchedNutriments.getValue(nutrient, option);
+            expect(
+              actual,
+              isNotNull,
+              reason: 'should not be null for $nutrient',
+            );
+            expect(
+              searchedNutriments.getValue(nutrient, perSize),
+              nutriments.getValue(nutrient, perSize),
+              reason: 'should be the same values for $nutrient',
+            );
+          }
+        }
+      }
+    }, skip: 'Works randomly');
+
+    test('Test saving food with PerSize.serving Nutrients', () async {
+      const User USER = TestConstants.TEST_USER;
+      const double ENERGY = 365;
+      const double CARBOHYDRATES = 12;
+      const double PROTEINS = 6;
+      const double FAT = 0.1;
+      const String BARCODE = '01234567891';
+      const String PRODUCT_NAME = 'Test Food';
+
+      for (final option in PerSize.values) {
```

Comment:
Bad naming: "option". What about "var1", "possibleValue"?
Jokes aside, `perSize` would be easier to read, wouldn't it?